nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -304,17 +306,16 @@ public void deleteComment(CoreSession session, String commentId)
         CoreInstance.doPrivileged(session, s -> {
             DocumentModel comment = s.getDocument(commentRef);
             String parentId = (String) comment.getPropertyValue(COMMENT_PARENT_ID);
-            DocumentRef parentRef = new IdRef(parentId);
-            DocumentRef ancestorRef = getAncestorRef(s, comment);
+            DocumentRef ancestorRef = getTopLevelCommentAncestor(s, commentRef);
             if (s.exists(ancestorRef) && !principal.isAdministrator()
                     && !comment.getPropertyValue(COMMENT_AUTHOR).equals(principal.getName())
                     && !s.hasPermission(principal, ancestorRef, SecurityConstants.EVERYTHING)) {
                 throw new CommentSecurityException(
                         "The user " + principal.getName() + " can not delete comments of document " + parentId);
```

Comment:
Shouldn't it be `ancestorRef` instead of `parentId`? (in which case `parentId` would become useless)

Reply:
you mean here: "The user " + principal.getName() + " can not delete comments of document " + **parentId**);

Reply:
yes, that's the line I've commented

Reply:
yes i see if i do it i am wondering if i don't break some changes on what we rely on the exception message -> but i will try it to see if id didn't break the tests

Reply:
i remove the parentId and replace it by the `ancestorRef` unfortunately this break some tests and i think it's better to keep like that to avoid any backward breaks and mainly change it, it' not really related to this ticket wdyt